lib/ldapUser.class.php

```diff
@@ -0,0 +1,165 @@
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+include sfConfig::get('sf_root_dir') .'/vendor/password_compat/password.php';
+
+class ldapUser extends myUser implements Zend_Acl_Role_Interface
+{
+  protected $ldapConnection;
+
+  public function authenticate($username, $password)
+  {
+    // Allow LDAP authentication to be overridden during development
+    $configuration = sfContext::getInstance()->getConfiguration();
+    if ($configuration->isDebug() || 'dev' == $configuration->getEnvironment())
+    {
+      return parent::authenticate($username, $password);
+    }
+
+    $authenticated = false;
+
+    // Anonymous is not a real user
+    if ($username == 'anonymous')
+    {
+      return false;
+    }
+
+    $authenticated = $this->ldapAuthenticate($username, $password);
+
+    // Fallback to non-LDAP authentication if need be and load/create user data
+    if (!$authenticated)
+    {
+      $authenticated = parent::authenticate($username, $password);
+
+      // Load user
+      $criteria = new Criteria;
+      $criteria->add(QubitUser::EMAIL, $username);
+      $user = QubitUser::getOne($criteria);
+    }
+    else
+    {
+      // Load user using username or, if one doesn't exist, create it
+      $criteria = new Criteria;
+      $criteria->add(QubitUser::USERNAME, $username);
+      if (null === $user = QubitUser::getOne($criteria))
+      {
+        $user = $this->createUserFromLdapInfo($username);
+      }
+    }
+
+    // Sign in user if authentication was successful
+    if ($authenticated)
+    {
+      $this->signIn($user);
+    }
+
+    return $authenticated;
+  }
+
+  protected function createUserFromLdapInfo($username)
+  {
+    $user = new QubitUser();
+    $user->username = $username;
+
+    // Do LDAP search for user's email address
+    $base_dn = (string)QubitSetting::getByName('ldapBaseDn');
+    $filter = '(uid='. $username .')';
+
+    $result = ldap_search($this->getLdapConnection(), $base_dn, $filter);
+    $entries = ldap_get_entries($this->getLdapConnection(), $result);
+
+    // If user is found and email exists, store it
+    if ($entries['count'] && !empty($entries[0]['mail']))
+    {
+      $user->email = $entries[0]['mail'][0];
+    }
+
+    $user->save();
+
+    return $user;
+  }
+
+  protected function getLdapConnection()
+  {
+    if (isset($this->ldapConnection))
+    {
+      return $this->ldapConnection;
+    }
+    else
+    {
+      $host = 'ldap://'. (string)QubitSetting::getByName('ldapHost');
+      $port = (string)QubitSetting::getByName('ldapPort');
+
+      $connection = ldap_connect($host, $port);
```

Comment:
As you are using full LDAP URIs you could as well add the port to it, e.g. `ldap_connect('ldap://hostname:port')` or you could `ldap_connect('localhost', 12345)`.